### DISCUSSION TITLE
How to pretty print JSON using `.Display()`

### DISCUSSION THREAD

**Discussion by NikiforovAll** [dotnet/interactive#3718] [Q&A] [answered]
The JSON is highlighted by not indented, I would like to be able to configure JsonSerilizer.

**Comment by jonsequitur** [comment]
Are you looking for [`JsonSerializerOptions.WriteIndented`](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.writeindented?view=net-8.0)?

**Comment by NikiforovAll** [comment]
Yes, but it is read-only

**Comment by jonsequitur** [comment]
Can you explain more about what you're trying to do?

**Reply by NikiforovAll** [reply]
I'm trying to print out the JSON in a formatted/highlighted manner for visualization purposes. I get the color highlighting, but the JSON is not written with indentation.

**Reply by jonsequitur** [reply, marked as answer]
Which JSON?

Code highlighting is a feature of VS Code and it will show up as long as the MIME type has a corresponding [notebook renderer](https://code.visualstudio.com/api/extension-guides/notebook#notebook-renderer).

![image](https://github.com/user-attachments/assets/d598e31f-72be-4d11-9cd5-abaa671ed40b)

Indentation is a feature of various JSON serializers. Here's a simple method to indent some JSON using System.Text.Json:

```csharp
    private static string Indent(string json)
    {
        json = JsonNode.Parse(json).ToJsonString(new JsonSerializerOptions
        {
            WriteIndented = true,
            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping
        });
        return json;
    }
```

**Reply by NikiforovAll** [reply]
Thank you, it works, my mistake is that I was passing the object instead of formated string
